coderd/database/queries/provisionerjobs.sql

```diff
@@ -22,13 +22,7 @@ WHERE
 			AND nested.organization_id = @organization_id
 			-- Ensure the caller has the correct provisioner.
 			AND nested.provisioner = ANY(@types :: provisioner_type [ ])
-			AND CASE
-				-- Special case for untagged provisioners: only match untagged jobs.
-				WHEN nested.tags :: jsonb = '{"scope": "organization", "owner": ""}' :: jsonb
-				THEN nested.tags :: jsonb = @tags :: jsonb
-				-- Ensure the caller satisfies all job tags.
-				ELSE nested.tags :: jsonb <@ @tags :: jsonb
-			END
+			AND tags_compatible(nested.tags, @tags :: jsonb)
```

Comment:
I think args are in wrong order here (given as [superset, subset])? But if you do the previously mentioned arg-swap from above, then they'll be correct here.

Reply:
Yeah, looks like you're right! I wonder if the original code is right though because this behavior confuses me. Doesn't this allow a job to have 3 tags and it can be picked up by a provisioner that only has 2 tags that match 2/3 for the job?

Maybe this is as intended but then I'm afraid I don't understand our tagging system 😄

Reply:
My latest commit should clarify the naming to make this more clear. Please check to see if it makes sense now?

Reply:
Now I see it yeah, my bad, I completely missed that it was a provisioner_JOB that was being nested. Looks good 👍🏻